src/main/java/com/enderio/base/common/item/tool/SoulVialItem.java

```diff
@@ -243,4 +260,73 @@ public MultiCapabilityProvider initCapabilities(ItemStack stack, @Nullable Compo
     }
 
     // endregion
+
+    // region events
+
+    /**
+     * Fix for certain mobs that don't work with the soul vial.
+     * So far the list includes:
+     * - Donkey
+     * - Mule
+     * - Llama
+     * - Villagers
+     */
+    @SubscribeEvent
+    public static void onLivingInteract(PlayerInteractEvent.EntityInteractSpecific event) {
+        ItemStack stack = event.getItemStack();
+        if (stack.is(EIOItems.EMPTY_SOUL_VIAL.get())) {
+            if (event.getTarget() instanceof AbstractChestedHorse || event.getTarget() instanceof Villager) {
+                stack.getItem().interactLivingEntity(stack, event.getPlayer(), (LivingEntity) event.getTarget(), event.getHand());
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onCommonSetup(FMLCommonSetupEvent event) {
```

Comment:
Can we move event handling to either a single place or a separate "SoulVialEventHandler" sort of thing? I think for code navigation that is simpler for finding all events and working them out together.

Reply:
I don't really know if anyone would really need to find all Events for understanding EnderIO.
The much likelier thing to happen is someone wants to understand how the Soul Vials work
Having some things in SoulVialItem and some in some EventListener multiple packages away doesn't really add much for code readability imo.

Reply:
I'm happy with that explanation. Maybe i'm old school when there were less events. This makes sense to me, happy to approve.